zuul-core/src/main/java/com/netflix/zuul/filters/endpoint/ProxyEndpoint.java

```diff
@@ -896,6 +846,10 @@ protected void handleOriginNonSuccessResponse(final HttpResponse originResponse,
                     startedSendingResponseToClient, zuulRequest.hasCompleteBody(), zuulRequest.getMethod());
             //detach from current origin.
             unlinkFromOrigin();
+
+            // ensure body reader indexes are reset so retry is able to access the body buffer
```

Comment:
Worth clarifying in this comment that we want Netty to have access to that ByteBuf, rather than Zuul itself.